src/backend/distributed/utils/node_metadata.c

```diff
@@ -478,10 +529,17 @@ RemoveNodeFromCluster(char *nodeName, int32 nodePort)
 		deletedNodeId = workerNode->nodeId;
 	}
 
-	DeleteNodeRow(nodeName, nodePort);
-
 	DeleteAllReferenceTablePlacementsFromNode(nodeName, nodePort);
 
+	hasAnyShardPlacements = NodeHasShardPlacements(nodeName, nodePort,
+												   onlyConsiderActivePlacements);
+	if (hasAnyShardPlacements)
+	{
+		ereport(ERROR, (errmsg("you cannot remove a node which has shard placements")));
```

Comment:
This is linked to the decision above, if we allow removing nodes while there are still placements then removing a node with an inactive placement gets the system into a state where queries start erroring out.